### PR TITLE
Using upstream docker over wily

### PR DESCRIPTION
**Pull request by fgimenez**
Also symlinked /var/lib/docker to /mnt to prevent exhausting the base disk
because of https://github.com/docker/docker/issues/3182

### CODE REVIEW

File: bin/remote/provision.sh [hunk 1]
```diff
@@ -2,7 +2,14 @@
 set -x
 
 install_docker(){
-    sudo apt-get install -y docker.io
+    sudo apt-key adv --keyserver hkp://p80.pool.sks-keyservers.net:80 --recv-keys 58118E89F3A912897C070ADBF76221572C52609D
+    echo "deb https://apt.dockerproject.org/repo ubuntu-wily main" | sudo tee /etc/apt/sources.list.d/docker.list
+    sudo apt-get update
+    sudo apt-get purge -y lxc-docker
+    sudo apt-get install -y linux-image-extra-$(uname -r) docker-engine
+    sudo mkdir -p /mnt/docker && sudo rm -rf /var/lib/docker && sudo ln -s /mnt/docker /var/lib/docker
```
Comment (elopio) [inline bin/remote/provision.sh]: A comment here would be nice.

**Comment by elopio** [comment]
Sorry, you might have told me this already, but what's wrong with docker from xenial?
As always in this branch, I trust you and I can't see anything wrong, so +1.

File: bin/remote/provision.sh (hunk 1 above)
Comment (fgimenez) [inline bin/remote/provision.sh]: Sure, done

**Comment by fgimenez** [comment]
> Sorry, you might have told me this already, but what's wrong with docker from xenial?

It's very old, in docker terms :) There have been a lot of improvements since 1.6 was out, including storage plugins, volume drivers and, most importantly, whole new networking. Moreover, thinking on the development based on kubernetes we have made so far, they have dropped the support for 1.6.2 https://github.com/kubernetes/kubernetes/pull/15994, which is exactly the version in xenial https://launchpad.net/ubuntu/xenial/+source/docker.io

**Comment by fgimenez** [comment]
Also, according to [1] installing the `linux-image-extra` package allows us to use the aufs storage driver, which prevents the errors that we were facing with devicemapper. Not sure if this is also true for the archive version.

[1] https://docs.docker.com/engine/installation/ubuntulinux/

**Comment by elopio** [comment]
ok, I knew that about the old version. What I found hard to understand is that the version from wily is newer than the version from wily. I suppose that's because we only have a ppa for wily right now, and then all makes sense :)
Thanks for the info.

**Pull request merged**